### DISCUSSION TITLE
Run Script with data from Event Rule

### DISCUSSION THREAD

**Discussion by PaulR282** [netbox-community/netbox#15227] [Help Wanted!]
I have a Script that needs to run every time an Interface is updated. I created an Event Rule that triggers every time an Interface is updated. How do you parse Information, like device and interface to the script? The Script has 2 ObjectVars (I could also build it with StringVars) for device and interface. I know I have to do it with the Action Data in the Event Rule but I don't know how to parse information from the event. Is this even possible?

**Comment by AADIP** [comment]
Information that is passed from the event rule is passed into `data`. So those objectvars that you normally use are not input when this script is ran from the event rule. What should be passed into `data` is the interface object that is being updated.

**Reply by PaulR282** [reply]
Thanks, I found all the information I need in `data`

**Reply by bitcollector1** [reply]
explain it like you are 5 please? I simply need to pass a device.serial into my script from the event but I'm too dense or tired to figure it out at the moment :( 

```
An exception occurred: KeyError: 'serial_num'

Traceback (most recent call last):
  File "/opt/netbox/netbox/extras/scripts.py", line 512, in _run_script
    script.output = script.run(data=data, commit=commit)
  File "/opt/netbox/netbox/scripts/netbox_curl_refresh.py", line 44, in run
    serial_num = data['serial_num']
KeyError: 'serial_num'

```

<img width="517" alt="Screenshot 2024-04-04 at 23 10 22" src="https://github.com/netbox-community/netbox/assets/50723251/9c4b81b2-92e3-4add-ab3e-c564a801d66a">

**Reply by PaulR282** [reply]
@bitcollector1 You don't need to do anything with the data field in the event rule. I don't think the serial will be directly parsed in the data object, so you need to do something like this:

```
from dcim.models import Device
device = Device.objects.filter(id=data["device"]["id"])
```
And then you can call device.serial

**Reply by bitcollector1** [reply]
thanks very much for the feedback! I was able to get the script to run just fine, but ran into a small problem where the script keeps running over and over again with a different username. it's really odd.....not entirely sure how or why this is happening :( 

![Screenshot 2024-04-05 at 09 41 49](https://github.com/netbox-community/netbox/assets/50723251/8d6041ca-e136-420b-8bad-2cf3ba900410)

**Reply by PaulR282** [reply]
> thanks very much for the feedback! I was able to get the script to run just fine, but ran into a small problem where the script keeps running over and over again with a different username. it's really odd.....not entirely sure how or why this is happening :( 
> 
> ![Screenshot 2024-04-05 at 09 41 49](https://github.com/netbox-community/netbox/assets/50723251/8d6041ca-e136-420b-8bad-2cf3ba900410)
> 

Are you sure you only have 1 event rule that triggers the script?

**Reply by bitcollector1** [reply]
yes, just the 1 event rule, this is my very first one.
<img width="1339" alt="Screenshot 2024-04-05 at 10 06 28" src="https://github.com/netbox-community/netbox/assets/50723251/8d7ed9c1-9010-4123-8b00-8703b0cb7ab3">
